File: core/application/src/main/java/com/gonezo/multiplatform/audioextraction/application/RequestGuard.java
package com.gonezo.multiplatform.audioextraction.application;

import com.gonezo.multiplatform.audioextraction.contract.ExtractionRequest;
import com.gonezo.multiplatform.audioextraction.contract.ExtractionResult;

public interface RequestGuard {
  void validateRequest(ExtractionRequest request);

  void validateResult(ExtractionResult result);
}

File: core/bin/main/com/gonezo/audioextraction/application/pipeline/RequestGuard.kt
package com.gonezo.audioextraction.application.pipeline

import com.gonezo.audioextraction.domain.contract.ExtractionRequest
import com.gonezo.audioextraction.domain.contract.ExtractionResult

interface RequestGuard {
  fun validateRequest(request: ExtractionRequest)
  fun validateResult(result: ExtractionResult)
}
